### PR TITLE
WRAP-UP: Task maze generation asset prototype

### PR DESCRIPTION
**Pull request by JCraft026**

### CODE REVIEW

File: Project Pathfinder/Assets/Scripts/GenerateMaze.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    // Generate all paths throughout the maze
+    private static WallStatus[,] GeneratePaths(WallStatus[,] mazeData, int mazeWidth, int mazeHeight){
+        Stack<CellPosition> positionStack = new Stack<CellPosition>(); // Holds maze cell position data
+        System.Random randomNum           = new System.Random();       // Random number generator
+        CellPosition initialCell          = new CellPosition{ _x = randomNum.Next(0, mazeWidth), _y = randomNum.Next(0, mazeHeight) },
+                                                                       // Starting cell for path generation
+                     currentCell          = new CellPosition(),        // Position of the current cell being processed
+                     neighborCell         = new CellPosition();        // Position of the selected neighbor of the current cell
+        List<CellNeighbor> cellNeighbors  = new List<CellNeighbor>();  // List of unvisited neighbors to the current cell
```
Comment (CalebFrancis) [inline Project Pathfinder/Assets/Scripts/GenerateMaze.cs]: could probably change "cellNeighbors" variable name to "unvisitedCellNeighbors" as it is a much clearer name

File: Project Pathfinder/Assets/Scripts/GenerateMaze.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                mazeData[neighborCell._x, neighborCell._y] |= WallStatus.VISITED;
+                positionStack.Push(neighborCell);
+            }
+        }
+        return mazeData;
+    }
+
+    // Get a list of the current cell's unvisited neighbors
+    private static List<CellNeighbor> GetUnvisitedNeighbours(CellPosition currentCell, WallStatus[,] mazeData, int mazeWidth, int mazeHeight){
+        List<CellNeighbor> cellNeighbors = new List<CellNeighbor>(); // List of unvisited neighbors to the current cell
```
Comment (CalebFrancis) [inline Project Pathfinder/Assets/Scripts/GenerateMaze.cs]: See previous comment about changing cellNeighbors to "unvisitedCellNeighbors"

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        WallStatus[,] mazeData = GenerateMaze.Generate(mazeWidth, mazeHeight);
+
+        // Render the maze in the scene
+        Render(mazeData);
+    }
+
+    // Render the complete maze within the scene
+    private void Render(WallStatus[,] mazeData){
+        WallStatus currentCell = new WallStatus(); // Current maze cell being rendered
+        Vector2 scenePosition  = new Vector2();    // x,y position in the scene
```
Comment (CalebFrancis) [inline Project Pathfinder/Assets/Scripts/RenderMaze.cs:38]: Maybe change "scenePosition" name to "wallPosition"

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        // Render the cell walls of every maze cell
+        for (int j = 0; j < mazeHeight; j++){
+            for (int i = 0; i < mazeWidth; i++){
+                currentCell = mazeData[i,j];
+                scenePosition = new Vector2(cellSize * (-mazeWidth / 2 + i), cellSize * (-mazeHeight / 2 + j));
+
+                // Render the top wall of a maze cell
+                if(currentCell.HasFlag(WallStatus.TOP)){
+                    var topWall        = Instantiate(wallPrefab, transform) as Transform;
```
Comment (CalebFrancis) [inline Project Pathfinder/Assets/Scripts/RenderMaze.cs:53]: Not a suggested change but just a question. Where is the "transform" variable coming from? I don't see it ever being created.

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                }
+            }
+        }
+            
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
```
Comment (CalebFrancis) [inline Project Pathfinder/Assets/Scripts/RenderMaze.cs]: We can probably remove Update() in the current iteration of this code.

**Review by CalebFrancis — CHANGES_REQUESTED** [review]
See comments regarding variable name clarity

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs (hunk 4 above)
Comment (JCraft026) [reply Project Pathfinder/Assets/Scripts/RenderMaze.cs:53]: Transform is a unity engine class, it is used here to make sure that the scale, rotation, and position values are cloned along with the base object

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs (hunk 3 above)
Comment (JCraft026) [reply Project Pathfinder/Assets/Scripts/RenderMaze.cs:38]: We should keep it as "scenePosition". In the future this variable will also be used to place items and other prefabs, not just cells. And this variable is storing a x,y scene coordinate value, its not storing a cell grid position, so calling it "cellPosition" would be confusing.

File: Project Pathfinder/Assets/Scripts/GenerateMaze.cs (hunk 1 above)
Comment (JCraft026) [reply Project Pathfinder/Assets/Scripts/GenerateMaze.cs]: Done.
changed it to "unvisitedNeighbors" so it matches the name of the function

File: Project Pathfinder/Assets/Scripts/GenerateMaze.cs (hunk 2 above)
Comment (JCraft026) [reply Project Pathfinder/Assets/Scripts/GenerateMaze.cs]: Done.
changed it to "unvisitedNeighbors" so it matches the name of the function

File: Project Pathfinder/Assets/Scripts/RenderMaze.cs (hunk 5 above)
Comment (JCraft026) [reply Project Pathfinder/Assets/Scripts/RenderMaze.cs]: Done

**Review by CalebFrancis — APPROVED** [review]
That works

**Pull request merged**